src/app/icd/ICDManager.cpp

```diff
@@ -154,6 +155,8 @@ void ICDManager::UpdateOperationState(OperationalState state)
             mOperationalState           = OperationalState::ActiveMode;
             uint32_t activeModeInterval = IcdManagementServer::GetInstance().GetActiveModeInterval();
             DeviceLayer::SystemLayer().StartTimer(System::Clock::Timeout(activeModeInterval), OnActiveModeDone, this);
+            DeviceLayer::SystemLayer().StartTimer(System::Clock::Timeout(activeModeInterval - ICD_ACTIVE_TIME_JITTER_MS),
```

Comment:
Are we starting one or two separate timers here ?

Reply:
We are starting two timers, one to inform us that we are about to go to idle mode, one to switch to idle mode. Note that depending on the system layer, this does not start two physical timer (In FreeRTOS for instance).